cunumeric/deferred.py

```diff
@@ -3224,3 +3223,36 @@ def unpackbits(
         scale = tuple(8 if dim == axis else 1 for dim in range(src.ndim))
         task.add_constraint(p_out <= p_in * scale)  # type: ignore
         task.execute()
+
+    @auto_convert([1])
+    def _wrap(self, src: Any, new_len: int) -> None:
+        if src.base.kind == Future:
```

Comment:
We should also apply any transformations that `src` might have, otherwise the copy will not work:
```
if src.base.kind == Future or src.base.transformed:
    src = self._convert_future_to_store(src)
```
and similarly for self:
```
if self.base.kind == Future or self.base.transformed:
    lhs = self._convert_future_to_store(self)
else:
    lhs = self
```
Note that I used a different variable than self, see below.

Reply:
thank you